modules/context/repo.go

```diff
@@ -407,6 +407,13 @@ func RepoIDAssignment() func(ctx *Context) {
 
 // RepoAssignment returns a middleware to handle repository assignment
 func RepoAssignment(ctx *Context) (cancel context.CancelFunc) {
+	repoAssignmentOnce, _ := ctx.Data["RepoAssignment"].(bool)
```

Comment:
Is `RepoAssignment` used anywhere else? If not, I prefer to use an internal name to indicate that this member field is only used inside `func RepoAssignment` and should not used anywhere else.

eg: `ctx.Data["repoAssignmentExecuted"]`, and we can use `if _, repoAssignmentOnce = ctx.Data["RepoAssignment"] {` to avoid the type casting